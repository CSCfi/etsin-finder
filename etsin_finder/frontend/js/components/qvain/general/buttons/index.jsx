export {
  TableButton,
  CancelButton,
  SaveButton,
  DangerButton,
  DangerCancelButton,
  RemoveButton,
  AddNewButton,
} from './commonButtons'

export {
  FileIcon,
  LinkIcon,
  ChevronRight,
  ChevronDown,
  EditButton,
  DeleteButton,
} from './iconButtons'

export { ButtonGroup, FileItem, ButtonContainer, ConfirmButtonContainer } from './buttonContainers'

export { ButtonLabel } from './buttonParts'
