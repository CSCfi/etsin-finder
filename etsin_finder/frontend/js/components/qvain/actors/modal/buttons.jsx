import React from 'react'
import PropTypes from 'prop-types'
import Translate from 'react-translate-component'

import { SaveButton, CancelButton } from '../../general/buttons'

const Buttons = ({ handleRequestClose, handleSaveActor, readonly }) => (
  <div style={{ marginTop: 'auto' }}>
    <Translate
      component={CancelButton}
      type="button"
      onClick={handleRequestClose}
      content="qvain.actors.add.cancel.label"
    />
    <Translate
      disabled={readonly}
      component={SaveButton}
      type="button"
      onClick={handleSaveActor}
      content="qvain.actors.add.save.label"
    />
  </div>
)

Buttons.propTypes = {
  handleRequestClose: PropTypes.func.isRequired,
  handleSaveActor: PropTypes.func.isRequired,
  readonly: PropTypes.bool.isRequired,
}

export default Buttons
