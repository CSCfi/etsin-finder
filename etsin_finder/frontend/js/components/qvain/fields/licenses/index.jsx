import React, { useState } from 'react'
import Translate from 'react-translate-component'
import translate from 'counterpart'
import AccessType from './accessType'
import Licenses from './licenses'
import { SectionTitle } from '../../general/section'
import Tooltip from '../../general/section/tooltip'
import { HelpIcon } from '../../general/modal/form'
import LicensesInfo from './licensesInfo'

const RightsAndLicenses = () => {
  const [tooltipOpen, setTooltipOpen] = useState(false)
  return (
    <div className="container">
      <SectionTitle>
        <Translate content="qvain.rightsAndLicenses.title" />
        <Tooltip
          isOpen={tooltipOpen}
          close={() => setTooltipOpen(!tooltipOpen)}
          align="Right"
          text={<LicensesInfo />}
        >
          <HelpIcon
            aria-label={translate('qvain.rightsAndLicenses.infoTitle')}
            onClick={() => setTooltipOpen(!tooltipOpen)}
          />
        </Tooltip>
      </SectionTitle>
      <Licenses />
      <AccessType />
    </div>
  )
}

export default RightsAndLicenses
