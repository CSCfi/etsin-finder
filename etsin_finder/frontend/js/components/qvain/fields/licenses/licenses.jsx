import React, { Component } from 'react'
import { inject, observer } from 'mobx-react'
import { withTheme } from 'styled-components'
import PropTypes from 'prop-types'
import Translate from 'react-translate-component'
import CreatableSelect from 'react-select/creatable'

import getReferenceData from '../../utils/getReferenceData'
import Card from '../../general/card'
import { LabelLarge } from '../../general/modal/form'
import { License as LicenseConstructor } from '../../../../stores/view/qvain'
import { licenseSchema } from '../../utils/formValidation'
import {
  onChangeMulti,
  getCurrentOption,
  getOptionLabel,
  getOptionValue,
  sortOptions,
  autoSortOptions,
} from '../../utils/select'
import { ValidationErrors } from '../../general/errors/validationError'

export class License extends Component {
  promises = []

  static propTypes = {
    Stores: PropTypes.object.isRequired,
    theme: PropTypes.object.isRequired,
  }

  state = {
    options: [],
    licenseErrors: {},
  }

  componentDidMount = () => {
    this.promises.push(
      getReferenceData('license')
        .then(res => {
          const list = res.data.hits.hits
          const options = list.map(ref => LicenseConstructor(ref._source.label, ref._source.uri))
          const { lang } = this.props.Stores.Locale
          sortOptions(LicenseConstructor, lang, options)
          this.setState({
            options,
          })
          autoSortOptions(this, this.props.Stores.Locale, LicenseConstructor)
        })
        .catch(error => {
          if (error.response) {
            // Error response from Metax
            console.log(error.response.data)
            console.log(error.response.status)
            console.log(error.response.headers)
          } else if (error.request) {
            // No response from Metax
            console.log(error.request)
          } else {
            // Something happened in setting up the request that triggered an Error
            console.log('Error', error.message)
          }
        })
    )
  }

  componentWillUnmount() {
    this.promises.forEach(promise => promise && promise.cancel && promise.cancel())
  }

  removeLicense = license => {
    this.props.Stores.Qvain.removeLicense(license)
  }

  validateLicenses = () => {
    const { licenseArray } = this.props.Stores.Qvain
    const licenseErrors = {}
    licenseArray.forEach(license => {
      const { identifier, name } = license
      const validationObject = { identifier, name, otherLicenseUrl: identifier }
      try {
        licenseSchema.validateSync(validationObject)
      } catch (err) {
        licenseErrors[identifier] = err.message
      }
    })
    this.setState({ licenseErrors })
  }

  onChange = values => {
    const { setLicenseArray } = this.props.Stores.Qvain
    onChangeMulti(setLicenseArray)(values)
    this.validateLicenses()
  }

  createLicense = url =>
    LicenseConstructor({ fi: `Muu (URL): ${url}`, en: `Other (URL): ${url}` }, url)

  render() {
    const { options, licenseErrors } = this.state
    const { lang } = this.props.Stores.Locale
    const { licenseArray, readonly } = this.props.Stores.Qvain

    // allow wrap for long license labels
    const styles = {
      multiValue: (style, state) => {
        if (this.state.licenseErrors[state.data.identifier]) {
          return {
            ...style,
            background: this.props.theme.color.error,
            color: 'white',
          }
        }
        return style
      },
      multiValueLabel: style => ({
        ...style,
        whiteSpace: 'normal',
        color: 'inherit',
      }),
    }

    return (
      <Card>
        <LabelLarge htmlFor="licenseSelect">
          <Translate content="qvain.rightsAndLicenses.license.title" />
        </LabelLarge>
        <Translate component="p" content="qvain.rightsAndLicenses.license.infoText" />
        <Translate
          component={CreatableSelect}
          inputId="licenseSelect"
          name="license"
          isDisabled={readonly}
          getOptionLabel={getOptionLabel(LicenseConstructor, lang)}
          getOptionValue={getOptionValue(LicenseConstructor)}
          value={getCurrentOption(LicenseConstructor, options, licenseArray)}
          options={options}
          isMulti
          isClearable={false}
          onChange={this.onChange}
          createOptionPosition="first"
          getNewOptionData={this.createLicense}
          attributes={{ placeholder: 'qvain.rightsAndLicenses.license.placeholder' }}
          styles={styles}
        />
        {licenseErrors && (
          <ValidationErrors
            errors={Object.entries(licenseErrors).map(([url, err]) => `${url}: ${err}`)}
          />
        )}
      </Card>
    )
  }
}

export default withTheme(inject('Stores')(observer(License)))
