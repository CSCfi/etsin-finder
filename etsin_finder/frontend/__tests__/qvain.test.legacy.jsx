import React from 'react'
import { shallow, mount } from 'enzyme'

import '../locale/translations'
import etsinTheme from '../js/styles/theme'
import Qvain from '../js/components/qvain/main'
import Description from '../js/components/qvain/description'
import DescriptionField from '../js/components/qvain/description/descriptionField'
import OtherIdentifierField from '../js/components/qvain/description/otherIdentifierField'
import FieldOfScienceField from '../js/components/qvain/description/fieldOfScienceField'
import KeywordsField from '../js/components/qvain/description/keywordsField'
import RightsAndLicenses from '../js/components/qvain/licenses'
import { License } from '../js/components/qvain/licenses/licenses'
import { AccessType } from '../js/components/qvain/licenses/accessType'
import RestrictionGrounds from '../js/components/qvain/licenses/restrictionGrounds'
import EmbargoExpires from '../js/components/qvain/licenses/embargoExpires'
import { ACCESS_TYPE_URL, LICENSE_URL, DATA_CATALOG_IDENTIFIER } from '../js/utils/constants'
import Files from '../js/components/qvain/files'
import IDAFilePicker, { IDAFilePickerBase } from '../js/components/qvain/files/legacy/idaFilePicker'
import FileSelector, { FileSelectorBase } from '../js/components/qvain/files/legacy/fileSelector'
import { SelectedFilesBase, FileLabel } from '../js/components/qvain/files/legacy/selectedFiles'
import { DeleteButton } from '../js/components/qvain/general/buttons'
import Env from '../js/stores/domain/env'
import QvainStoreClass, {
  Directory,
  AccessType as AccessTypeConstructor,
  License as LicenseConstructor,
} from '../js/stores/view/qvain'
import LocaleStore from '../js/stores/view/language'

jest.mock('uuid', () => {
  let id = 0
  return {
    v4: () => id++,
  }
})

const QvainStore = new QvainStoreClass(Env)
const getStores = () => {
  Env.setMetaxApiV2(false)
  return {
    Env,
    Qvain: QvainStore,
    Locale: LocaleStore,
  }
}

describe('Qvain', () => {
  it('should render correctly', () => {
    const component = shallow(<Qvain Stores={getStores()} />)

    expect(component).toMatchSnapshot()
  })

  it('should open existing dataset when the identifier in the url changes', () => {
    const stores = getStores()

    // Mock react router matches for identifier
    const emptyMatch = { params: { identifier: null } }
    const identifierMatch = { params: { identifier: 'some_identifier' } }
    const anotherMatch = { params: { identifier: 'another_identifier' } }

    // Replace Qvain.getDataset so we can test it was called correctly
    let callCount = 0
    let lastCall
    class FakeQvain extends Qvain.WrappedComponent.wrappedComponent {
      getDataset(identifier) {
        callCount += 1
        lastCall = identifier
      }
    }

    // Create empty dataset, getDataset should not be called
    shallow(
      <FakeQvain Stores={stores} match={emptyMatch} history={{}} location={{ pathname: '/' }} />
    )
    expect(callCount).toBe(0)
    expect(lastCall).toBe(undefined)

    // Dataset already opened for editing (e.g. in the dataset view), getDataset should not be called
    lastCall = undefined
    const datasetOpenedStore = {
      ...stores,
      Qvain: {
        ...stores.Qvain,
        original: { identifier: identifierMatch.params.identifier },
      },
    }
    shallow(
      <FakeQvain
        Stores={datasetOpenedStore}
        match={identifierMatch}
        history={{}}
        location={{ pathname: '/' }}
      />
    )
    expect(callCount).toBe(0)
    expect(lastCall).toBe(undefined)

    // Different dataset already opened, getDataset should be called with the new identifier
    lastCall = undefined
    const anotherDatasetOpenedStore = {
      ...stores,
      Qvain: {
        ...stores.Qvain,
        original: { identifier: anotherMatch.params.identifier },
      },
    }
    shallow(
      <FakeQvain
        Stores={anotherDatasetOpenedStore}
        match={identifierMatch}
        history={{}}
        location={{ pathname: '/' }}
      />
    )
    expect(callCount).toBe(1)
    expect(lastCall).toBe(identifierMatch.params.identifier)

    // Edit existing dataset, getDataset should be called
    lastCall = undefined
    const wrapper = shallow(
      <FakeQvain
        Stores={stores}
        match={identifierMatch}
        history={{}}
        location={{ pathname: '/' }}
      />
    )
    expect(callCount).toBe(2)
    expect(lastCall).toBe(identifierMatch.params.identifier)

    // Switch to another dataset, getDataset should be called
    lastCall = undefined
    wrapper.setProps({ match: anotherMatch })
    expect(callCount).toBe(3)
    expect(lastCall).toBe(anotherMatch.params.identifier)
  })
})

describe('Qvain.Description', () => {
  it('should render <Description />', () => {
    const component = shallow(<Description Stores={getStores()} />)
    expect(component).toMatchSnapshot()
  })
  it('should render <DescriptionField />', () => {
    const component = shallow(<DescriptionField Stores={getStores()} />)
    expect(component).toMatchSnapshot()
  })
  it('should render <OtherIdentifierField />', () => {
    const component = shallow(<OtherIdentifierField Stores={getStores()} />)
    expect(component).toMatchSnapshot()
  })
  it('should render <FieldOfScienceField />', () => {
    const component = shallow(<FieldOfScienceField Stores={getStores()} />)
    expect(component).toMatchSnapshot()
  })
  it('should render <KeywordsField />', () => {
    const component = shallow(<KeywordsField Stores={getStores()} />)
    expect(component).toMatchSnapshot()
  })
})

describe('Qvain.RightsAndLicenses', () => {
  it('should render <RightsAndLicenses />', () => {
    const component = shallow(<RightsAndLicenses />)
    expect(component).toMatchSnapshot()
  })
  it('should render <Licenses />', () => {
    const stores = getStores()
    const component = shallow(<License Stores={stores} theme={etsinTheme} />)
    expect(component).toMatchSnapshot()
  })
  it('should render <AccessType />', () => {
    const stores = getStores()
    const component = shallow(<AccessType Stores={stores} />)
    expect(component).toMatchSnapshot()
  })
  it('should render <RestrictionGrounds />', () => {
    const stores = getStores()
    stores.Qvain.setAccessType(AccessTypeConstructor(undefined, ACCESS_TYPE_URL.EMBARGO))
    const component = shallow(<AccessType Stores={stores} />)
    expect(component.find(RestrictionGrounds).length).toBe(1)
  })
  it('should NOT render <RestrictionGrounds />', () => {
    const stores = getStores()
    stores.Qvain.setAccessType(AccessTypeConstructor(undefined, ACCESS_TYPE_URL.OPEN))
    const component = shallow(<AccessType Stores={stores} />)
    expect(component.find(RestrictionGrounds).length).toBe(0)
  })
  it('should render <EmbargoExpires />', () => {
    const stores = getStores()
    stores.Qvain.setAccessType(AccessTypeConstructor(undefined, ACCESS_TYPE_URL.EMBARGO))
    const component = shallow(<AccessType Stores={stores} />)
    expect(component.find(EmbargoExpires).length).toBe(1)
  })
  it('should NOT render <EmbargoExpires />', () => {
    const stores = getStores()
    stores.Qvain.setAccessType(AccessTypeConstructor(undefined, ACCESS_TYPE_URL.OPEN))
    const component = shallow(<AccessType Stores={stores} />)
    expect(component.find(EmbargoExpires).length).toBe(0)
  })
})

describe('Qvain.Files', () => {
  it('should render file picker', () => {
    const store = getStores()
    store.Qvain.dataCatalog = DATA_CATALOG_IDENTIFIER.IDA
    store.Qvain.idaPickerOpen = true
    const component = shallow(<Files Stores={store} />)
    expect(component.dive().find(IDAFilePicker).length).toBe(1)
  })

  it('should open file selector upon selecting file picker', () => {
    const component = shallow(<IDAFilePickerBase Stores={getStores()} />)
    component
      .children()
      .last()
      .simulate('click', {
        preventDefault: () => console.log('preventDefault'),
      })
    expect(component.find(FileSelector).length).toBe(1)
  })

  it('allows selecting directories in the file selector', () => {
    const stores = getStores()
    const component = mount(<FileSelectorBase Stores={stores} />)
    stores.Qvain.selectedProject = 'project_y'
    stores.Qvain.hierarchy = Directory(
      {
        id: 'test1',
        identifier: 'test-ident-1',
        project_identifier: 'project_y',
        directory_name: 'root',
        directories: [
          Directory(
            {
              id: 'test2',
              identifier: 'test-ident-2',
              project_identifier: 'project_y',
              directory_name: 'directory2',
              directories: [],
              files: [],
            },
            undefined,
            false,
            false
          ),
        ],
        files: [],
      },
      undefined,
      false,
      true
    )
    component.update()
    expect(component.find('li').length).toBe(1)
    component.find('li').find('input').simulate('change')
    expect(component.props().Stores.Qvain.hierarchy.directories[0].selected)
  })

  it('allows modifying the metadata of selected directories', () => {
    // repeat previous one
    const stores = getStores()
    // reset selected directories
    stores.Qvain.selectedDirectories = []
    const fileSelector = mount(<FileSelectorBase Stores={stores} />)

    stores.Qvain.selectedProject = 'project_y'
    stores.Qvain.hierarchy = Directory(
      {
        id: 'test1',
        identifier: 'test-ident-1',
        project_identifier: 'project_y',
        directory_name: 'root',
        directories: [
          {
            id: 'test2',
            identifier: 'test-ident-2',
            project_identifier: 'project_y',
            directory_name: 'directory2',
            directories: [],
            files: [],
          },
        ],
        files: [],
      },
      undefined,
      false,
      true
    )
    fileSelector.update()
    fileSelector.find('#test2Checkbox input').simulate('change')
    fileSelector.unmount()
    // mount the SelectedFiles component
    const selectedFiles = shallow(<SelectedFilesBase Stores={stores} />)
    expect(selectedFiles.find(FileLabel).last().text()).toBe(
      '<FontAwesomeIcon />project_y / directory2'
    )
    selectedFiles
      .find(DeleteButton)
      .last()
      .simulate('click', { preventDefault: () => {} })
    expect(selectedFiles.find(FileLabel).length).toBe(0)
  })
})
