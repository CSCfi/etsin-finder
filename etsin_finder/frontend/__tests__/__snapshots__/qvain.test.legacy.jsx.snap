// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qvain should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Qvain.Description should render <Description /> 1`] = `
<div
  className="container"
>
  <title__SectionTitle>
    <Translate
      component="span"
      content="qvain.description.title"
    />
    <Tooltip
      align="Right"
      close={[Function]}
      isOpen={false}
      text={<DescriptionInfo />}
    >
      <HelpIcon
        aria-label="Description info"
        onClick={[Function]}
      />
    </Tooltip>
  </title__SectionTitle>
  <inject-DescriptionField-with-Stores />
  <inject-IssuedDateField-with-Stores />
  <inject-OtherIdentifierField-with-Stores />
  <inject-FieldOfScienceField-with-Stores />
  <inject-KeywordsField-with-Stores />
</div>
`;

exports[`Qvain.Description should render <DescriptionField /> 1`] = `
<DescriptionField
  Stores={
    Object {
      "Locale": Locale {},
      "Qvain": Qvain {
        "Actors": Actors {
          "Qvain": [Circular],
          "actorInEdit": null,
          "actors": Array [],
          "createActor": [Function],
          "editDataset": [Function],
          "getDatasetOrganizations": [Function],
          "getReferenceOrganizations": [Function],
          "getReferenceOrganizationsForActor": [Function],
          "isActorEqual": [Function],
          "isEqual": [Function],
          "isOrganizationEqual": [Function],
          "loadingReferenceOrganizations": Object {},
          "referenceOrganizationErrors": Object {},
          "referenceOrganizations": Object {},
          "reset": [Function],
          "toBackend": [Function],
        },
        "Files": Files {
          "AddItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "Qvain": [Circular],
          "SelectedItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "getExisting": [Function],
          "getItemByPath": [Function],
          "inEdit": undefined,
          "loadingProject": null,
          "refreshModalDirectory": null,
          "root": null,
          "selectedExistingCounter": ChildItemCounter {
            "root": Object {
              "count": 0,
              "directories": Object {},
            },
          },
          "selectedExistingMetadata": Object {},
          "selectedProject": undefined,
          "toMetax": [Function],
        },
        "accessType": Object {
          "name": undefined,
          "url": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open",
        },
        "changed": false,
        "createExternalResourceUIId": [Function],
        "cumulativeState": 0,
        "dataCatalog": undefined,
        "deprecated": false,
        "description": Object {
          "en": "",
          "fi": "",
        },
        "deselectChildren": [Function],
        "deselectParents": [Function],
        "embargoExpDate": undefined,
        "existingDirectories": Array [],
        "existingFiles": Array [],
        "extResFormOpen": false,
        "externalResourceInEdit": Object {
          "accessUrl": "",
          "downloadUrl": "",
          "id": undefined,
          "title": "",
          "useCategory": "",
        },
        "externalResources": Array [],
        "fieldOfScience": undefined,
        "fieldsOfScience": Array [],
        "fixDeprecatedModalOpen": false,
        "hierarchy": Object {},
        "idaPickerOpen": false,
        "inEdit": undefined,
        "infrastructure": undefined,
        "infrastructures": Array [],
        "issuedDate": undefined,
        "keywords": Array [],
        "legacyFilePicker": true,
        "license": Object {
          "identifier": "http://uri.suomi.fi/codelist/fairdata/license/code/CC-BY-4.0",
          "name": undefined,
        },
        "mergeArraysByIdentifier": [Function],
        "metadataModalFile": undefined,
        "original": undefined,
        "otherIdentifiers": Array [],
        "otherLicenseUrl": undefined,
        "preservationState": 0,
        "restrictionGrounds": Object {},
        "selectedDirectories": Array [],
        "selectedFiles": Array [],
        "selectedProject": undefined,
        "title": Object {
          "en": "",
          "fi": "",
        },
        "useDoi": false,
      },
    }
  }
/>
`;

exports[`Qvain.Description should render <FieldOfScienceField /> 1`] = `
<FieldOfScienceField
  Stores={
    Object {
      "Locale": Locale {},
      "Qvain": Qvain {
        "Actors": Actors {
          "Qvain": [Circular],
          "actorInEdit": null,
          "actors": Array [],
          "createActor": [Function],
          "editDataset": [Function],
          "getDatasetOrganizations": [Function],
          "getReferenceOrganizations": [Function],
          "getReferenceOrganizationsForActor": [Function],
          "isActorEqual": [Function],
          "isEqual": [Function],
          "isOrganizationEqual": [Function],
          "loadingReferenceOrganizations": Object {},
          "referenceOrganizationErrors": Object {},
          "referenceOrganizations": Object {},
          "reset": [Function],
          "toBackend": [Function],
        },
        "Files": Files {
          "AddItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "Qvain": [Circular],
          "SelectedItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "getExisting": [Function],
          "getItemByPath": [Function],
          "inEdit": undefined,
          "loadingProject": null,
          "refreshModalDirectory": null,
          "root": null,
          "selectedExistingCounter": ChildItemCounter {
            "root": Object {
              "count": 0,
              "directories": Object {},
            },
          },
          "selectedExistingMetadata": Object {},
          "selectedProject": undefined,
          "toMetax": [Function],
        },
        "accessType": Object {
          "name": undefined,
          "url": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open",
        },
        "changed": false,
        "createExternalResourceUIId": [Function],
        "cumulativeState": 0,
        "dataCatalog": undefined,
        "deprecated": false,
        "description": Object {
          "en": "",
          "fi": "",
        },
        "deselectChildren": [Function],
        "deselectParents": [Function],
        "embargoExpDate": undefined,
        "existingDirectories": Array [],
        "existingFiles": Array [],
        "extResFormOpen": false,
        "externalResourceInEdit": Object {
          "accessUrl": "",
          "downloadUrl": "",
          "id": undefined,
          "title": "",
          "useCategory": "",
        },
        "externalResources": Array [],
        "fieldOfScience": undefined,
        "fieldsOfScience": Array [],
        "fixDeprecatedModalOpen": false,
        "hierarchy": Object {},
        "idaPickerOpen": false,
        "inEdit": undefined,
        "infrastructure": undefined,
        "infrastructures": Array [],
        "issuedDate": undefined,
        "keywords": Array [],
        "legacyFilePicker": true,
        "license": Object {
          "identifier": "http://uri.suomi.fi/codelist/fairdata/license/code/CC-BY-4.0",
          "name": undefined,
        },
        "mergeArraysByIdentifier": [Function],
        "metadataModalFile": undefined,
        "original": undefined,
        "otherIdentifiers": Array [],
        "otherLicenseUrl": undefined,
        "preservationState": 0,
        "restrictionGrounds": Object {},
        "selectedDirectories": Array [],
        "selectedFiles": Array [],
        "selectedProject": undefined,
        "title": Object {
          "en": "",
          "fi": "",
        },
        "useDoi": false,
      },
    }
  }
/>
`;

exports[`Qvain.Description should render <KeywordsField /> 1`] = `
<KeywordsField
  Stores={
    Object {
      "Locale": Locale {},
      "Qvain": Qvain {
        "Actors": Actors {
          "Qvain": [Circular],
          "actorInEdit": null,
          "actors": Array [],
          "createActor": [Function],
          "editDataset": [Function],
          "getDatasetOrganizations": [Function],
          "getReferenceOrganizations": [Function],
          "getReferenceOrganizationsForActor": [Function],
          "isActorEqual": [Function],
          "isEqual": [Function],
          "isOrganizationEqual": [Function],
          "loadingReferenceOrganizations": Object {},
          "referenceOrganizationErrors": Object {},
          "referenceOrganizations": Object {},
          "reset": [Function],
          "toBackend": [Function],
        },
        "Files": Files {
          "AddItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "Qvain": [Circular],
          "SelectedItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "getExisting": [Function],
          "getItemByPath": [Function],
          "inEdit": undefined,
          "loadingProject": null,
          "refreshModalDirectory": null,
          "root": null,
          "selectedExistingCounter": ChildItemCounter {
            "root": Object {
              "count": 0,
              "directories": Object {},
            },
          },
          "selectedExistingMetadata": Object {},
          "selectedProject": undefined,
          "toMetax": [Function],
        },
        "accessType": Object {
          "name": undefined,
          "url": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open",
        },
        "changed": false,
        "createExternalResourceUIId": [Function],
        "cumulativeState": 0,
        "dataCatalog": undefined,
        "deprecated": false,
        "description": Object {
          "en": "",
          "fi": "",
        },
        "deselectChildren": [Function],
        "deselectParents": [Function],
        "embargoExpDate": undefined,
        "existingDirectories": Array [],
        "existingFiles": Array [],
        "extResFormOpen": false,
        "externalResourceInEdit": Object {
          "accessUrl": "",
          "downloadUrl": "",
          "id": undefined,
          "title": "",
          "useCategory": "",
        },
        "externalResources": Array [],
        "fieldOfScience": undefined,
        "fieldsOfScience": Array [],
        "fixDeprecatedModalOpen": false,
        "hierarchy": Object {},
        "idaPickerOpen": false,
        "inEdit": undefined,
        "infrastructure": undefined,
        "infrastructures": Array [],
        "issuedDate": undefined,
        "keywords": Array [],
        "legacyFilePicker": true,
        "license": Object {
          "identifier": "http://uri.suomi.fi/codelist/fairdata/license/code/CC-BY-4.0",
          "name": undefined,
        },
        "mergeArraysByIdentifier": [Function],
        "metadataModalFile": undefined,
        "original": undefined,
        "otherIdentifiers": Array [],
        "otherLicenseUrl": undefined,
        "preservationState": 0,
        "restrictionGrounds": Object {},
        "selectedDirectories": Array [],
        "selectedFiles": Array [],
        "selectedProject": undefined,
        "title": Object {
          "en": "",
          "fi": "",
        },
        "useDoi": false,
      },
    }
  }
/>
`;

exports[`Qvain.Description should render <OtherIdentifierField /> 1`] = `
<OtherIdentifierField
  Stores={
    Object {
      "Locale": Locale {},
      "Qvain": Qvain {
        "Actors": Actors {
          "Qvain": [Circular],
          "actorInEdit": null,
          "actors": Array [],
          "createActor": [Function],
          "editDataset": [Function],
          "getDatasetOrganizations": [Function],
          "getReferenceOrganizations": [Function],
          "getReferenceOrganizationsForActor": [Function],
          "isActorEqual": [Function],
          "isEqual": [Function],
          "isOrganizationEqual": [Function],
          "loadingReferenceOrganizations": Object {},
          "referenceOrganizationErrors": Object {},
          "referenceOrganizations": Object {},
          "reset": [Function],
          "toBackend": [Function],
        },
        "Files": Files {
          "AddItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "Qvain": [Circular],
          "SelectedItemsView": DirectoryView {
            "Files": [Circular],
            "checkedState": Object {},
            "getTopmostChecked": [Function],
            "isOpen": [Function],
            "openState": Object {},
            "showAllState": Object {},
            "showLimit": 20,
            "showLimitMargin": 1,
          },
          "getExisting": [Function],
          "getItemByPath": [Function],
          "inEdit": undefined,
          "loadingProject": null,
          "refreshModalDirectory": null,
          "root": null,
          "selectedExistingCounter": ChildItemCounter {
            "root": Object {
              "count": 0,
              "directories": Object {},
            },
          },
          "selectedExistingMetadata": Object {},
          "selectedProject": undefined,
          "toMetax": [Function],
        },
        "accessType": Object {
          "name": undefined,
          "url": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open",
        },
        "changed": false,
        "createExternalResourceUIId": [Function],
        "cumulativeState": 0,
        "dataCatalog": undefined,
        "deprecated": false,
        "description": Object {
          "en": "",
          "fi": "",
        },
        "deselectChildren": [Function],
        "deselectParents": [Function],
        "embargoExpDate": undefined,
        "existingDirectories": Array [],
        "existingFiles": Array [],
        "extResFormOpen": false,
        "externalResourceInEdit": Object {
          "accessUrl": "",
          "downloadUrl": "",
          "id": undefined,
          "title": "",
          "useCategory": "",
        },
        "externalResources": Array [],
        "fieldOfScience": undefined,
        "fieldsOfScience": Array [],
        "fixDeprecatedModalOpen": false,
        "hierarchy": Object {},
        "idaPickerOpen": false,
        "inEdit": undefined,
        "infrastructure": undefined,
        "infrastructures": Array [],
        "issuedDate": undefined,
        "keywords": Array [],
        "legacyFilePicker": true,
        "license": Object {
          "identifier": "http://uri.suomi.fi/codelist/fairdata/license/code/CC-BY-4.0",
          "name": undefined,
        },
        "mergeArraysByIdentifier": [Function],
        "metadataModalFile": undefined,
        "original": undefined,
        "otherIdentifiers": Array [],
        "otherLicenseUrl": undefined,
        "preservationState": 0,
        "restrictionGrounds": Object {},
        "selectedDirectories": Array [],
        "selectedFiles": Array [],
        "selectedProject": undefined,
        "title": Object {
          "en": "",
          "fi": "",
        },
        "useDoi": false,
      },
    }
  }
/>
`;

exports[`Qvain.RightsAndLicenses should render <AccessType /> 1`] = `
<Card>
  <form__LabelLarge
    htmlFor="accessTypeSelect"
  >
    <Translate
      component="span"
      content="qvain.rightsAndLicenses.accessType.title"
    />
  </form__LabelLarge>
  <Translate
    attributes={
      Object {
        "placeholder": "qvain.rightsAndLicenses.accessType.placeholder",
      }
    }
    clearable={true}
    component={[Function]}
    inputId="accessTypeSelect"
    isDisabled={false}
    name="accessType"
    onBlur={[Function]}
    onChange={[Function]}
    options={Array []}
    value={
      Object {
        "label": undefined,
        "value": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open",
      }
    }
  />
  <validationError__ValidationError />
</Card>
`;

exports[`Qvain.RightsAndLicenses should render <Licenses /> 1`] = `
<Card>
  <form__LabelLarge
    htmlFor="licenseSelect"
  >
    <Translate
      component="span"
      content="qvain.rightsAndLicenses.license.title"
    />
  </form__LabelLarge>
  <Translate
    attributes={
      Object {
        "placeholder": "qvain.rightsAndLicenses.license.placeholder",
      }
    }
    component={[Function]}
    inputId="licenseSelect"
    isClearable={true}
    isDisabled={false}
    name="license"
    onBlur={[Function]}
    onChange={[Function]}
    options={Array []}
    value={
      Object {
        "label": undefined,
        "value": undefined,
      }
    }
  />
</Card>
`;

exports[`Qvain.RightsAndLicenses should render <RightsAndLicenses /> 1`] = `
<div
  className="container"
>
  <title__SectionTitle>
    <Translate
      component="span"
      content="qvain.rightsAndLicenses.title"
    />
    <Tooltip
      align="Right"
      close={[Function]}
      isOpen={false}
      text={<LicensesInfo />}
    >
      <HelpIcon
        aria-label="Rights And Licenses info"
        onClick={[Function]}
      />
    </Tooltip>
  </title__SectionTitle>
  <inject-License-with-Stores />
  <inject-AccessType-with-Stores />
</div>
`;
